pkg/controller/beat/common/pod_test.go

```diff
@@ -158,6 +262,7 @@ func Test_buildPodTemplate(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			tt.args.params.Context = context.Background()
```

Comment:
If we are defaulting this here, which is good, I think we should remove it where it is set in the test cases.